### PR TITLE
Add new functions get_path_to_protocol and render_protocol

### PR DESCRIPTION
**Pull request by hansvancalster**
Addresses [this comment](https://github.com/inbo/protocolhelper/pull/9#discussion_r432294696)
Also fixing functions `get_short_titles` and `get_protocolnumbers` which no longer worked due to new design choices.

### CODE REVIEW

**Comment by ThierryO** [comment]
Closes #15 

The syntax above will automatically close the references issue when the PR is merged with master

File: R/create_protocol.R [hunk 1]
```diff
@@ -71,8 +71,7 @@
 #' the target folder beneath `src`.
 #' Besides Rmarkdown files, this target folder will also contain files needed to render
 #'  to a Bookdown gitbook such as a `_bookdown.yml`.
-#' Additionally, a `NEWS.Rmd` file will be written to the target folder which
```
Comment (ThierryO) [inline R/create_protocol.R:74]: I'd keep the NEWS.Rmd filename. And set the files to render (and their order) via `_bookdown.yml`. See https://bookdown.org/yihui/bookdown/usage.html

File: R/create_protocol.R [hunk 2]
```diff
@@ -397,23 +387,86 @@ get_short_titles <- function(
...
+      subfolder_of <- "thematic"
+      protocol_leading_number <- themes_df[themes_df$theme == theme,
+                                           "theme_number"]
+      theme <- paste0(protocol_leading_number, "_", theme)
+      path_to_protocol <- file.path(project_root,
+                                    "src",
+                                    subfolder_of,
+                                    theme,
+                                    protocol_folder_name)
+    } else {
```
Comment (ThierryO) [inline R/create_protocol.R:460]: you can use an early return in this case. That make longer if else statements more reable.

```
if(condition) {
    # do something when TRUE
    return(output) # returns output and ends the function
}
# do something else. You only get here when condition == FALSE
return(output)
```

File: R/create_protocol.R [hunk 3]
```diff
@@ -397,23 +387,86 @@ get_short_titles <- function(
...
+
+  # first case: the path exists already
+  project_root <- find_root(is_git_root)
+  ld <- list.dirs(path = file.path(project_root,"src"),
+                  full.names = TRUE,
+                  recursive = TRUE)
+  ld <- str_subset(string = ld,
+                   pattern = protocol_folder_name)
+  if (!identical(ld, character(0))) {
+    path_to_protocol <- ld[[1]]
```
Comment (ThierryO) [inline R/create_protocol.R:442]: use an early return (see next comment).

File: R/render_all.R [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+#'
+#'
+#' @examples
+#' \dontrun{
+#' render_all(version_number = "2000.01")
+#'}
+
+
+render_all <- function(output_root = "publish",
+                       version_number = NULL) {
```
Comment (ThierryO) [inline R/render_all.R:36]: don't set a default for `version_number`. Then you don't need the test of it being `NULL`.

File: R/render_all.R [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+  assert_that(is.string(output_root))
+  assert_that(!is.null(version_number),
+              str_detect(version_number, "^\\d{4}\\.\\d{2}$"),
+              msg = "Please provide a valid version number")
+
+  git_root <- find_root(is_git_root)
+  output_new_root <- path(git_root, output_root, version_number)
```
Comment (ThierryO) [inline R/render_all.R:44]: This won't work. The folder `YYYY.NN` should only contain the protocol that was updated during that version. Otherwise the repository would contain too much redundant information.

Comment (hansvancalster) [reply R/render_all.R:44]: Good point. So all the function needs to do is:

- render the protocol which has that specific `YYYY.NN` tag (possibly more than one in case of multiple languages). The protocol-specific NEWS.Rmd file should reflect which changes have been made.
- automatically update the repo NEWS.Rmd - which gives an overview of all versions of all protocols with links to them and render this NEWS.Rmd into a homepage #19

?

File: R/render_all.R (hunk 4 above)
Comment (hansvancalster) [reply R/render_all.R]: See 8131735

File: R/create_protocol.R (hunk 3 above)
Comment (hansvancalster) [reply R/create_protocol.R:442]: See 8131735

File: R/create_protocol.R (hunk 2 above)
Comment (hansvancalster) [reply R/create_protocol.R:462]: See 8131735

File: R/render_all.R (hunk 5 above)
Comment (ThierryO) [reply R/render_all.R:43]: Maybe the homepage should only list the procotols, not every version of the protocol. So you could list all protocol folders in `publish`

**Comment by hansvancalster** [comment]
> Here are a few suggestions. It would be handy if we have a dummy repo with a few fake protocols to test this.

I've made a fork of the protocols repo in my account and added some dummy protocols. See [fork](https://github.com/hansvancalster/protocols).

**Comment by ThierryO** [comment]
> I've made a fork of the protocols repo in my account and added some dummy protocols. See [fork](https://github.com/hansvancalster/protocols).

Can you give me admin rights to that repo. Then I can setup the CI on that repo to test it.

**Comment by hansvancalster** [comment]
> > I've made a fork of the protocols repo in my account and added some dummy protocols. See [fork](https://github.com/hansvancalster/protocols).
> 
> Can you give me admin rights to that repo. Then I can setup the CI on that repo to test it.

Yes, but I think you need to accept the invite first.

**Comment by hansvancalster** [comment]
> > > I've made a fork of the protocols repo in my account and added some dummy protocols. See [fork](https://github.com/hansvancalster/protocols).
> > 
> > 
> > Can you give me admin rights to that repo. Then I can setup the CI on that repo to test it.
> 
> Yes, but I think you need to accept the invite first.

It appears I cannot set "roles" for collaborators on my account... is that a problem to test CI on that repo?

**Comment by ThierryO** [comment]
> > > > I've made a fork of the protocols repo in my account and added some dummy protocols. See [fork](https://github.com/hansvancalster/protocols).
> > > 
> > > 
> > > Can you give me admin rights to that repo. Then I can setup the CI on that repo to test it.
> > 
> > 
> > Yes, but I think you need to accept the invite first.
> 
> It appears I cannot set "roles" for collaborators on my account... is that a problem to test CI on that repo?

I'll create my own fork if it is a problem.

File: R/render_all.R (hunk 5 above)
Comment (florisvdh) [reply R/render_all.R:44]: Will `YYYY.NN` effectively become the _git tag_? I ask that because you use the 'tag' term above and also in `render_all()` documentation.

- If so, there is contradiction with the tag names of the branching model, where `protocols-YYYY.NN` is used for clarity. Also, I think that git tags should start with a letter for GitHub to be happy.
- If not, better not call these version numbers 'tags' (but just: version numbers, [see readme](https://github.com/inbo/protocols#version-number)) to prevent confusion.

Comment (hansvancalster) [reply R/render_all.R:44]: Good catch. Fixed this by 1b96dff
